Expanded search icon

vguzzi opened the pull request:
Adds expanded search icon. Search icon touch functions the same as the 'Search' soft keyboard button. Search icon only visible if search box has text inside of it.

vguzzi committed:
Adds expanded search icon. Search icon touch functions the same as the 'Search' soft keyboard button. Search icon only visible if search box has text inside of it.

sahildave commented on searchviewlayout/src/main/java/xyz/sahildave/widget/SearchViewLayout.java:
```diff
@@ -142,6 +166,19 @@ public void onClick(View v) {
+        mExpandedSearchIcon.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Editable editable = mSearchEditText.getText();
```
This and Line 129 can be extracted into a method.

vguzzi committed:
Extracts search functionality into search method.

vguzzi committed:
Fixed bug where soft keyboard would display 'done' and not search when pressed.
Changes IME options of search EditText to IME_ACTION_SEARCH. Soft keyboard now displays a search icon in place of done and also displays 'SEARCH' text when phone is in landscape 'IME mode'. Changes onEditorAction listener to check for IME_ACTION_SEARCH instead of IME_ACTION_NEXT to capture search press.

vguzzi commented:
Hi sahildave. This is my first open source project which I've worked on so please let me know if I'm doing anything wrong on the Github side of things. Still learning about pull requests etc on Github. Let me know if there's something specific you want me to look into!

sahildave commented:
Merged. Thanks for the pull request and bug fix.
